### PR TITLE
Update to GraalVM / Mandrel 22.3.0

### PR DESCRIPTION
**Pull request by cescoffier**
DO NOT MERGE!

### CODE REVIEW

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -656,7 +656,7 @@ Sample multistage Dockerfile for building an image from `scratch`:
 [source, dockerfile]
 ----
 ## Stage 1 : build with maven builder image with native capabilities
-FROM quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:22.2-java11 AS build
+FROM quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:22.3-java11 AS build
```
Comment (zakkak) [inline docs/src/main/asciidoc/building-native-image.adoc:659]: This should be:

```suggestion
FROM quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:{graalvm-flavor} AS build
```

**Review by zakkak — DISMISSED** [review]
Needs rebasing now that https://github.com/quarkusio/quarkus/pull/28843 got merged.

**Comment by gsmet** [comment]
Let me rebase it.

**Comment by gsmet** [comment]
Done! We still need to adjust the Mandrel image version when available but I updated the doc version as it won't break anything.

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
 @ConfigRoot(phase = ConfigPhase.BUILD_TIME)
 public class NativeConfig {
 
-    public static final String DEFAULT_GRAALVM_BUILDER_IMAGE = "quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:22.2-java17";
+    public static final String DEFAULT_GRAALVM_BUILDER_IMAGE = "quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:22.3-java17";
     public static final String DEFAULT_MANDREL_BUILDER_IMAGE = "quay.io/quarkus/ubi-quarkus-mandrel-builder-image:22.2-java17";
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:20]: We need to update that one when ready.

Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:20]: To make sure we won't forget: we need to update that one when ready.

**Review by gsmet — CHANGES_REQUESTED** [review]

**Comment by gsmet** [comment]
@zakkak I see the build failing with:
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project quarkus-core: Compilation failure: Compilation failure: 
Error:  /home/runner/work/quarkus/quarkus/core/runtime/src/main/java/io/quarkus/runtime/graal/ConfigurationSubstitutions.java:[6,36] cannot find symbol
Error:    symbol:   class AlwaysInline
Error:    location: package com.oracle.svm.core.annotate
Error:  /home/runner/work/quarkus/quarkus/core/runtime/src/main/java/io/quarkus/runtime/graal/ConfigurationSubstitutions.java:[26,6] cannot find symbol
Error:    symbol:   class AlwaysInline
Error:    location: class io.quarkus.runtime.graal.Target_io_smallrye_config_SmallRyeConfigProviderResolver
Error:  -> [Help 1]
```

Did we miss something when ensuring `22.3` compatibility?

**Comment by zakkak** [comment]
> Did we miss something when ensuring `22.3` compatibility?

Hmmm, indeed. Unfortunately Quarkus uses the defined `graalvm-version` for the GraalVM artifacts no matter which GraalVM version we use. So in our tests javac was building with the 22.2 GraalVM jars despite using 22.3 to build the actual native image. IIRC @Sanne experienced this at some point as well.

Regarding this specific failure, it's because [`com.oracle.svm.core.annotate.AlwaysInline` (which is not public API) was moved to `com.oracle.svm.core.AlwaysInline`](https://github.com/oracle/graal/pull/4683/files#diff-e9ce1eb4db69a014a3366a094cffd77696668c25de9094c6628168a0e9193bda).

So it looks like we will need to patch the source code and bump the minimum version to 22.3.

**Comment by gsmet** [comment]
I pushed an update to move to the new package. Rationale explained in the commit comment.

**Comment by cescoffier** [comment]
@zakkak unfortunately, it cannot be backported, at least not directly.
These images use the new multi-archs and jdock models. We need to use the former model for the backport.

**Comment by zakkak** [comment]
> @zakkak unfortunately, it cannot be backported, at least not directly. These images use the new multi-archs and jdock models. We need to use the former model for the backport.

Oh, I see. Shall we keep the label though to indicate that we need this in 2.13 as well?

**Comment by gsmet** [comment]
Yes, let's keep the label so that we don't miss it.

**Comment by gsmet** [comment]
@cescoffier what's the status of this build in your fork? Did you see other problems?

**Comment by cescoffier** [comment]
Nope, looks good.
The 2.13 images have been produced. Waiting for mandrel.

**Comment by zakkak** [comment]
> Waiting for mandrel.

FYI, due to some infrastructure related issues this might take a bit longer than usual.

**Comment by zakkak** [comment]
Mandrel 22.3 is now released. I have also openned https://github.com/quarkusio/quarkus-images/pull/215 to generate the quarkus images. @cescoffier are you going to update this PR to include the Mandrel images as well?

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java (hunk 2 above)
Comment (jerboaa) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:20]: `quay.io/quarkus/ubi-quarkus-mandrel-builder-image:22.3-java17` is there now, fwiw.

**Review by zakkak — APPROVED** [review]
LGTM, thanks @cescoffier

**Comment by gsmet** [comment]
I rebased to trigger CI as it was in limbo.

**Review by gsmet — APPROVED** [review]

**Comment by rsvoboda** [comment]
I see errors in our daily runs, it's apparently because of https://github.com/oracle/graal/issues/5303 bug

Shouldn't we set mandrel image as default for now? Peter has commented on the issue that Mandrel 22.3 works for him.

**Comment by zakkak** [comment]
@gsmet I am marking this as a "noteworthy" feature for better visibility.

(inspired by discussion in https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Quarkus.20won't.20build.20native/near/307716675)

**Comment by zakkak** [comment]
@rsvoboda I really don't know. I personally have not seen this issue happen in any of the machines I have access to.

I would suggest commenting on the upstream issue and possibly providing more info about the context under which the issue is triggered.

Regarding switching to Mandrel by default I am +0. People getting hit by the issue can easily switch to Mandrel by passing `-Dquarkus.native.builder-image=mandrel`. If it turns out to be a pretty common issue I expect upstream to make a new release soonish.

**Comment by rsvoboda** [comment]
ok, I will comment upstream

**Pull request merged**